packages/components/rating/tests/rating.test.tsx

```diff
@@ -126,4 +126,44 @@ describe("<Rating />", () => {
     await user.click(items[4]!)
     expect(items[3]).not.toHaveAttribute("data-focus")
   })
+
+  test("should use custom color correctly", () => {
+    const getColor = (value: number) => {
+      switch (value) {
+        case 1:
+          return "red.500"
+        case 2:
+          return "orange.500"
+        case 3:
+          return "yellow.500"
+        case 4:
+          return "green.500"
+        case 5:
+          return "blue.500"
+        default:
+          return undefined
+      }
+    }
```

Comment:
How about writing the code like this? (This is just my opinion :))

```tsx
  test.each([1, 2, 3, 4, 5])(
    "should use color for defaultValue %i",
    (defaultValue) => {
      const colors: Record<number, string> = {
        1: "red.500",
        2: "orange.500",
        3: "yellow.500",
        4: "green.500",
        5: "blue.500",
      }

      const getColor = (value: number): string | undefined => colors[value]

      // Below is the existing code.

      const { container } = render(
        <Rating color={getColor} defaultValue={defaultValue} />,
      )
  )
```

Reply:
Thank you for the idea! However, when I used `test.each`, it failed on tests since different `defaultValue` is applied across the tests. Each test renders the component with a different `defaultValue`, it affects how the `data-filled` attributes and CSS classes. 

But I like the object type to handle color values and applied it on the tests 🙌

Reply:
And using `Record<number, string>` didn't pass the ESlint, so I used the index signature.